File: lib/src/models/forum_recruitment_detail.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/forum_recruitment_intensity_label.dart';
import '../enums/forum_recruitment_tone_label.dart';
import 'general_user.dart';

part 'forum_recruitment_detail.g.dart';

@JsonSerializable()
class ForumRecruitmentDetail{
	
	ForumRecruitmentDetail();

	factory ForumRecruitmentDetail.fromJson(Map<String, dynamic> json) => _$ForumRecruitmentDetailFromJson(json);

	@JsonKey(name:'topicId')
	String topicId;
	@JsonKey(name:'microphoneRequired')
	bool microphoneRequired;
	@JsonKey(name:'intensity',unknownEnumValue:ForumRecruitmentIntensityLabel.None)
	ForumRecruitmentIntensityLabel intensity;
	@JsonKey(name:'tone',unknownEnumValue:ForumRecruitmentToneLabel.None)
	ForumRecruitmentToneLabel tone;
	@JsonKey(name:'approved')
	bool approved;
	@JsonKey(name:'conversationId')
	String conversationId;
	@JsonKey(name:'playerSlotsTotal')
	int playerSlotsTotal;
	@JsonKey(name:'playerSlotsRemaining')
	int playerSlotsRemaining;
	@JsonKey(name:'Fireteam')
	List<GeneralUser> fireteam;
	@JsonKey(name:'kickedPlayerIds')
	List<String> kickedPlayerIds;

	
	
	Map<String, dynamic> toJson() => _$ForumRecruitmentDetailToJson(this);
}

File: pkg/models/ForumRecruitmentDetail.go
package bungieapigo

type ForumRecruitmentDetail struct {
	TopicId              int64                          `json:"topicId"`
	MicrophoneRequired   bool                           `json:"microphoneRequired"`
	Intensity            ForumRecruitmentIntensityLabel `json:"intensity"`
	Tone                 ForumRecruitmentToneLabel      `json:"tone"`
	Approved             bool                           `json:"approved"`
	ConversationId       int64                          `json:"conversationId"`
	PlayerSlotsTotal     int                            `json:"playerSlotsTotal"`
	PlayerSlotsRemaining int                            `json:"playerSlotsRemaining"`
	Fireteam             []GeneralUser                  `json:"Fireteam"`
	KickedPlayerIds      []int64                        `json:"kickedPlayerIds"`
}
